commons/com.b2international.index/src/com/b2international/index/es/EsDocumentSearcher.java

```
@@ -183,6 +185,9 @@ public EsDocumentSearcher(EsIndexAdmin admin, ObjectMapper mapper) {
 		try {
 			response = client.search(req);
 		} catch (Exception e) {
+			if (e instanceof ElasticsearchStatusException && ((ElasticsearchStatusException) e).status() == RestStatus.BAD_REQUEST) {
+				throw new IllegalArgumentException(e.getMessage(), e);
+			}
```

Please add this block to a separate `catch (ElasticSearchStatusException e) { ... }` clause above the existing "catch-all" block.

---

Leaving as is now, as the current catch-all block handles the logging and generic index exception in case of any other ES exception thrown by the search request.

---

Following discussion with @cmark, non-400 errors should still be logged, so the `catch` block can be left as is.